src/main/java/com/easypost/model/Shipment.java

```diff
@@ -976,6 +976,18 @@ public Shipment buy(final boolean withCarbonOffset, final String apiKey) throws
         return this.buy(new HashMap<String, Object>() {}, withCarbonOffset, apiKey);
     }
 
+    /**
+     * Buy this Shipment.
+     *
+     * @param endShipperId the id of the end shipper to use for this purchase.
+     * @param apiKey       API key to use in request (overrides default API key).
+     * @return Shipment object
+     * @throws EasyPostException when the request fails.
+     */
+    public Shipment buy(final String endShipperId, final String apiKey) throws EasyPostException {
```

Comment:
Issue: We will never be able to use this function because a rate **must** be passed to buy a shipment.

Reply:
We need to overhaul `buy` in Shipment, there are a few `buy` functions that do not have `rate` or `param` in the parameters. For example: https://github.com/EasyPost/easypost-java/blob/master/src/main/java/com/easypost/model/Shipment.java#L918-L931

Reply:
fixed, but I believe we need to remove/deprecate other unnecessary functions in a separate PR

Reply:
Right, we need to remove any override buy function that doesn't contain a `rate` parameter because it will error out at the API because no rate was passed.

Let's not add an unusable override in this PR. Since we are fixing the buys here, you can remove any buy override function that does not have a rate and add a CHANGELOG entry stating we `removed unusable shipment buy functions that did not contain a rate parameter`

Reply:
It's called overload :)